concourse/pipelines/create-bosh-cloudfoundry.yml

```diff
@@ -449,7 +464,67 @@ jobs:
             passed: ['generate-bosh-config']
             trigger: true
           - get: cf-secrets
-
+          - get: metrics-cert
+          - get: vpc-tfstate
+          - get: concourse-tfstate
+          - get: bosh-tfstate
+          - get: vpc-terraform-state
+      - task: vpc-terraform-outputs-to-sh
+        config:
+          platform: linux
+          image: docker:///ruby#2.2-slim
+          inputs:
+          - name: paas-cf
+          - name: vpc-terraform-state
+          outputs:
+          - name: vpc-terraform-state
+            path: terraform-outputs
+          run:
+            path: sh
+            args:
+            - -e
+            - -c
+            - |
+              ruby paas-cf/concourse/scripts/extract_tf_vars_from_terraform_state.rb \
+              < vpc-terraform-state/vpc.tfstate \
+              > terraform-outputs/tfvars.sh
+              ls -l terraform-outputs/tfvars.sh
+              . terraform-outputs/tfvars.sh
+      - task: generate-metrics-certificates
+        config:
+          platform: linux
+          image: docker:///governmentpaas/curl-ssl
+          inputs:
+            - name: metrics-cert
+              path: existing-metrics-cert
+            - name: vpc-terraform-state
+          outputs:
+            - name: generated-metrics-cert
+          params:
+            DEPLOY_ENV: {{deploy_env}}
+          run:
+            path: sh
+            args:
+              - -e
+              - -c
+              - |
+                . vpc-terraform-state/tfvars.sh
+                env
+                ls -alR
+                tar xzvf existing-metrics-cert/metrics-cert.tar.gz
+                 if [ -f metrics.crt ] && [ -f metrics.key ]; then
+                   echo Certificate and private key already created, nothing to do
+                   cp existing-metrics-cert/metrics-cert.tar.gz generated-metrics-cert/.
+                   exit 0
+                 fi
+                 openssl req -x509 -newkey rsa:2048 -keyout metrics.key \
+                   -out metrics.crt -days 365 -nodes -subj \
+                   "/C=UK/ST=London/L=London/O=GDS/CN=metrics.{{deploy_env}}.${TF_VAR_system_dns_zone_name}"
```

Comment:
In other PR #159, we completely get rid of the variable `system_dns_zone_name`. Instead, there is a variable called `$ENV_DNS_ZONE` passed to concourse as the parameter `{{env_dns_zone_name}}` which you can use.

Using the variable `{{env_dns_zone_name}}` will actually remove the need to extract the `TF_VAR` in the previous task. 

See https://github.com/alphagov/paas-cf/pull/159/files#diff-5eac5d6fc041d1da10144e10c0fa2ed0R261 as an example